Samples/StyleTransfer/VideoEffect/StyleTransferEffectComponent/StyleTransferVideoEffect.cs

```diff
@@ -0,0 +1,236 @@
+        public void SetEncodingProperties(VideoEncodingProperties encodingProperties, IDirect3DDevice device)
+        {
+            this.encodingProperties = encodingProperties;
+        }
+
+        public IReadOnlyList<VideoEncodingProperties> SupportedEncodingProperties
+        {
+            get
+            {
+                var encodingProperties = new VideoEncodingProperties();
+                encodingProperties.Subtype = "ARGB32";
+                return new List<VideoEncodingProperties>() { encodingProperties };
+            }
+        }
+        public MediaMemoryTypes SupportedMemoryTypes { get { return MediaMemoryTypes.GpuAndCpu; } }
+
+        public bool TimeIndependent { get { return true; } }
+
+        private IPropertySet configuration;
+        public void SetProperties(IPropertySet configuration)
+        {
+            this.configuration = configuration;
+        }
+
+        public LearningModelBinding Binding
+        {
+            get
+            {
+                object val;
+                if (configuration != null && configuration.TryGetValue("Binding", out val))
+                {
+                    return (LearningModelBinding)val;
+                }
+                return null;
+            }
+        }
+
+        public LearningModelSession Session
+        {
+            get
+            {
+                object val;
+                if (configuration != null && configuration.TryGetValue("Session", out val))
+                {
+                    return (LearningModelSession)val;
+                }
+                return null;
+            }
+        }
+
+        public string InputImageDescription
+        {
+            get
+            {
+                object val;
+                if (configuration != null && configuration.TryGetValue("InputImageDescription", out val))
+                {
+                    return (string)val;
+                }
+                return null;
+            }
+        }
+
+        public string OutputImageDescription
+        {
+            get
+            {
+                object val;
+                if (configuration != null && configuration.TryGetValue("OutputImageDescription", out val))
+                {
+                    return (string)val;
+                }
+                return null;
+            }
+        }
+
+
+        public void ProcessFrame(ProcessVideoFrameContext context)
+        {
+            using (VideoFrame inputVideoFrame = context.InputFrame)
+            using (VideoFrame outputVideoFrame = context.OutputFrame)
+            {
+                _binding = Binding;
+                _session = Session;
+                _inputImageDescription = InputImageDescription;
+                _outputImageDescription = OutputImageDescription;
+
+                Task.Run(async () =>
+                {
+                    VideoFrame inputTransformed = await CenterCropImageAsync(inputVideoFrame, 720, 720);

```

Comment:
>720, 720 [](start = 94, length = 8)

probably okay for now, but add a comment to indicate that this probaby shoudl be inferred and not hard coded